backend/src/main/java/de/wirvsvirus/kek/service/locations/controller/LocationDataController.java

```diff
@@ -0,0 +1,66 @@
+package de.wirvsvirus.kek.service.locations.controller;
+
+import de.wirvsvirus.kek.service.diary.model.Diary;
+import de.wirvsvirus.kek.service.diary.repository.DiaryRepository;
+import de.wirvsvirus.kek.service.locations.model.LocationMatch;
+import de.wirvsvirus.kek.service.locations.model.pojo.TimeLineObject;
+import de.wirvsvirus.kek.service.locations.model.pojo.TimelineJsonRoot;
+import de.wirvsvirus.kek.service.locations.repository.LocationHistory;
+import de.wirvsvirus.kek.service.locations.repository.LocationHistoryRepository;
+import de.wirvsvirus.kek.service.locations.service.TrivialLocationMappingService;
+import io.swagger.annotations.ApiOperation;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.ResponseEntity;
+import org.springframework.web.bind.annotation.*;
+
+import java.util.Collection;
+import java.util.List;
+import java.util.stream.Collectors;
+
+@RestController
+@RequestMapping("/")
+public class LocationDataController {
+
+    @Autowired
+    private LocationHistoryRepository locationHistoryRepository;
+
+    @Autowired
+    private TrivialLocationMappingService trivialLocationMappingService;
+
+    @PostMapping("/locations/{user}/upload")
+    @ApiOperation(value = "Responds with a list of diaries, if parameters are set it will respond with a list of contacts taken between start and finish", response = Diary.class)
```

Comment:
response shouldn't be Diary here ;) would guess you could just remove the response specification for now as spring should automatically derive the response type from the method return type